test/asn1_encode_test.c

```diff
@@ -709,15 +709,18 @@ static int do_encode_custom(EXPECTED *input,
 static int do_print_item(const TEST_PACKAGE *package)
 {
 #define DATA_BUF_SIZE 256
-    unsigned char buf[DATA_BUF_SIZE];
     const ASN1_ITEM *i = ASN1_ITEM_ptr(package->asn1_type);
-    ASN1_VALUE *o = (ASN1_VALUE *)&buf;
+    ASN1_VALUE *o;
     int ret;
 
     OPENSSL_assert(package->encode_expectations_elem_size <= DATA_BUF_SIZE);
+    if ((o = OPENSSL_malloc(DATA_BUF_SIZE)) == NULL)
+        return 0;
 
-    (void)RAND_bytes(buf, (int)package->encode_expectations_elem_size);
+    (void)RAND_bytes((unsigned char*)o,
```

Comment:
I wonder if the result code of RAND_bytes should be checked?

Reply:
RAND is just a side-effect of the test, and this PR just fixes the alignment issue.  Perhaps a new PR to go check all uses and fix them?

(Besides, I'm about to push this :)

Reply:
There's no need to check the output of RAND_bytes() here. We don't care what data is placed in "buf", that's why RAND_bytes is used.  We could equally memset() the buffer to zero. We're testing the templates, not the data.